.github/workflows/ci.yml

```diff
@@ -186,7 +186,7 @@ jobs:
 
       - name: Generate build system
         run: |
-          cmake -B build --preset vs2022-static -DCMAKE_TOOLCHAIN_FILE="$env:VCPKG_INSTALLATION_ROOT\scripts\buildsystems\vcpkg.cmake" -DBUILD_BENCH=ON -DBUILD_FUZZ_BINARY=ON -DWERROR=ON
+          cmake -B build --preset vs2022-static -DCMAKE_TOOLCHAIN_FILE="$env:VCPKG_INSTALLATION_ROOT\scripts\buildsystems\vcpkg.cmake" -DWITH_BDB=ON -DWITH_MINIUPNPC=ON -DWITH_ZMQ=ON -DBUILD_BENCH=ON -DBUILD_FUZZ_BINARY=ON -DWERROR=ON
```

Comment:
From https://github.com/bitcoin/bitcoin/actions/runs/10665138943/job/29557888413#step:8:1:
```bash
Preset CMake variables:
  BUILD_GUI="ON"
  VCPKG_TARGET_TRIPLET="x64-windows-static"
  WITH_NATPMP="OFF"
  WITH_QRENCODE="OFF"
```
Can you explain why some options are passed on the command line, and why some are put into `CMakePresets.json`; what determines where each option belongs? I'd think it'd be better to have a single source of truth for the CI config.

Also, I see [compared to 28.x](https://github.com/bitcoin/bitcoin/blob/28.x/build_msvc/vcpkg.json), we've added new packages since porting to CMake, i.e `miniupnpc`, so why not also add [libqrencode](https://github.com/microsoft/vcpkg/blob/master/ports/libqrencode/vcpkg.json), or is there a reason it's hardcoded to OFF?

Reply:
For example, I can imagine someone changes `CMakePresets.json`, without realising that also partially controls the CI, and then we (silently) lose Qt compilation in this build.

Reply:
`WITH_NATPMP` is `OFF` because the `libnatpmp` package is not available in [vcpkg](https://vcpkg.io/en/packages).

`WITH_QRENCODE` is `OFF` because I faced build errors for this [package](https://vcpkg.io/en/package/libqrencode) or its dependencies in the past (I can't recall the exact issue). I've tested it recently and I got another build error, which is weird, because it happens in the main build system:
```
 Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
 ```
 It needs to be investigated further.

Reply:
> WITH_NATPMP is OFF
> WITH_QRENCODE is OFF

It seems like for at least `libnatpmp`, it'd be better to just set nothing, rather than hardcoding to `OFF`. As a package that doesn't exist can't cause issues, and I assume it could exist at some point, or the user could vendor it themselves, and then they would just need to override this (it also can't cause issues unless we opt into using it, even if it somehow ended up installed).

For libqrencode, if it's going to cause build issues, if the user has it and Qt installed, it'd probably be good to document that as the reason it is disabled.

Reply:
> For libqrencode, if it's going to cause build issues, if the user has it and Qt installed, it'd probably be good to document that as the reason it is disabled.

In suggestions about the place for such a note. Asking because the JSON format has no comments, and CMake's parser does not allow extra fields, such as `"_comment:"`.

Reply:
> In suggestions about the place for such a note.

I guess somewhere in the Windows build docs? Note, still wondering in regards to:

> Can you explain why some options are passed on the command line, and why some are put into CMakePresets.json; what determines where each option belongs? I'd think it'd be better to have a single source of truth for the CI config.

Reply:
Presets contain options that seem reasonable in most cases.

Reply:
> Presets contain options that seem reasonable in most cases.

Sure, but that's too vauge to use as reasoning for where CI configuration should live, and doesn't solve this issue:

> For example, I can imagine someone changing CMakePresets.json, without realising that also partially controls the CI, and then we (silently) lose Qt compilation in this build.

I'm not sure the configuration of the CI, should be (unexplicitly) coupled to the CMake Preset. This isn't specific to the GUI option, it holds for all options. Developers modifying these files need to know what lives where, and why. If we want to use Presets for the CI, that's possibly also fine, but they should be dedicated presets, so it's clear what they are for.

Reply:
> If we want to use Presets for the CI, that's possibly also fine, but they should be dedicated presets, so it's clear what they are for.

You highlighted a problem that is relevant to all CI jobs and is wider than this PR goal. Can we leave it for a follow up dedicated discussion/PR?

Reply:
Seems like we could just add the relevant options from the preset to the ci config file for now, which makes the config clear, and removes the possibility for external breakage? If at some point we switch to presets, they would all them be moved to a dedicated CI preset.

Reply:
> > In suggestions about the place for such a note.
> 
> I guess somewhere in the Windows build docs?

Thanks! Done.

> Seems like we could just add the relevant options from the preset to the ci config file for now

Implemented.